src/server/IoDispatchers.cpp

```diff
@@ -269,6 +269,118 @@ static bool _shouldAttemptHandoff(const Globals& globals,
 #endif
 }
 
+static void attemptHandoff(Globals& Globals, const CONSOLE_INFORMATION& gci, CONSOLE_API_CONNECTINFO& cac, PCONSOLE_API_MSG pReceiveMsg)
+{
+    if (!_shouldAttemptHandoff(Globals, gci, cac))
+    {
+        return;
+    }
+
+    try
+    {
+        // Go get ourselves some COM.
+        auto coinit = wil::CoInitializeEx(COINIT_MULTITHREADED);
+
+        // Get the class/interface to the handoff handler. Local machine only.
+        ::Microsoft::WRL::ComPtr<IConsoleHandoff> handoff;
+        THROW_IF_FAILED(CoCreateInstance(Globals.handoffConsoleClsid.value(), nullptr, CLSCTX_LOCAL_SERVER, IID_PPV_ARGS(&handoff)));
+
+        if (Globals.defaultTerminalMarkerCheckRequired)
+        {
+            ::Microsoft::WRL::ComPtr<IDefaultTerminalMarker> marker;
+            if (FAILED(handoff.As(&marker)))
+            {
+                Globals.handoffConsoleClsid.reset();
+                Globals.defaultTerminalMarkerCheckRequired = false;
+                return;
+            }
+        }

```

Comment:
note: this is the new blob of code here

Reply:
Yep the remaining function is unchanged. It was extracted into a standalone function so that I can write `return;` within this new if condition.